lib/shopify_api_extensions/backoff.rb

```diff
@@ -29,7 +29,7 @@ def request(*args)
       end
 
       # TODO look at code instead of static string? This is brittle
-      if e.class == ActiveResource::ClientError && e.message != "Failed.  Response code = 429.  Response message = Too Many Requests."
+      if e.class == ActiveResource::ClientError && e.message == "Failed.  Response code = 429.  Response message = Too Many Requests."
```

Comment:
I don't think this is correct. We only want to run the hit the rate limit retry logic if the exception **is** the rate limit error.

I took a look at the docs here:

http://www.rubydoc.info/gems/activeresource/4.1.0/ActiveResource/ClientError

And it doesn't look like `ActiveResource::ClientError` exposes a response code. `e.response` might have a response code exposed, if it does we'll want to do something like:

```
if e.class == ActiveResource::ClientError && e.code == 429 && message =~ /Too Many Requests/
```